src/Apps/MyCollection/Routes/MyCollectionArtwork/Components/MyCollectionArtworkSWASection.tsx

```diff
@@ -1,149 +1,134 @@
-              subject: "Submit This Artwork to Sell",
-              platform: "web",
-            })
-          }}
-        >
-          Submit for Sale
-        </Button>
-      </RouterLink>
+  const handleSubmitArtwork = async () => {
+    tracking.trackEvent({
+      contextModule: ContextModule.topWorksRail,
+      context_page_owner_type: OwnerType.myCollectionArtwork,
+      context_page_owner_id: artwork.internalID,
+      context_page_owner_slug: artwork?.artist?.slug,
+      destination_page_owner_type: OwnerType.consignmentFlow,
+      subject: "Submit This Artwork to Sell",
+      platform: "web",
+    })
 
-      <Text color="black60" variant="xs">
-        Learn more about{" "}
-        <Clickable
-          onClick={() => {
-            learnMore()
-            tracking.trackEvent({
-              contextModule: ContextModule.topWorksRail,
-              context_page_owner_type: OwnerType.myCollectionArtwork,
-              context_page_owner_id: artworkId,
-              context_page_owner_slug: slug,
-              subject: "Learn More",
-              platform: "web",
-            })
-          }}
-          color="black60"
-          textDecoration="underline"
-          data-testid="learn-more"
-        >
-          selling with Artsy.
-        </Clickable>
-      </Text>
-      <Separator my={2} />
-    </>
-  )
-}
+    if (!enableNewSubmissionFlow) return
 
-export const MyCollectionArtworkSWASectionDesktopLayout: React.FC<Props> = ({
-  route,
-  learnMore,
-  slug,
-  artworkId,
-  ctaColor = "primaryBlack",
-}) => {
-  const tracking = useTracking()
+    setIsLoading(true)
+
+    try {
+      let submissionID = artwork.consignmentSubmission?.internalID
+
+      if (!submissionID) {
+        submissionID = await createOrUpdateConsignSubmission(relayEnvironment, {
+          sessionID: !isLoggedIn ? getENV("SESSION_ID") : undefined,
+          artistID: artwork.artist?.internalID,
+          myCollectionArtworkID: artwork.internalID,
+          // Source is necessary in order to link this to a mycollection artwork
+          source: "MY_COLLECTION",
+        })
+      }
+
+      router.push(`/sell2/submissions/${submissionID}/artist`)
+    } catch (error) {
+      console.error(error)
+    } finally {
+      setIsLoading(false)
+    }
+  }
 
   return (
-    <>
-      <Separator my={2} />
+    <Box my={2}>
       <Text mb={0.5} variant="sm-display">
         Interested in Selling This Work?
       </Text>
+
       <Text mb={2} color="black60" variant="xs">
         Let our experts find the best sales option for you.
       </Text>
-      <RouterLink
-        textDecoration="none"
-        to={route}
+
+      <Button
+        variant="primaryBlack"
+        width="100%"
+        mb={2}
+        // @ts-ignore
+        as={enableNewSubmissionFlow ? undefined : RouterLink}
+        to={
+          !enableNewSubmissionFlow &&
+          `/collector-profile/my-collection/submission/artwork-details/${artwork.internalID}`
```

Comment:
we can probably remove this route as well once we release

Reply:
Yes, we can remove it in the end. I've already created a ticket for it: https://artsyproduct.atlassian.net/browse/ONYX-1074

Reply:
awesome